clients/src/main/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumer.java

```diff
@@ -1226,6 +1236,8 @@ private void close(Duration timeout, boolean swallowException) {
         closeTimer.update();
         if (applicationEventHandler != null)
             closeQuietly(() -> applicationEventHandler.close(Duration.ofMillis(closeTimer.remainingMs())), "Failed shutting down network thread", firstException);
+        swallow(log, Level.ERROR, "Failed invoking asynchronous commit callback.", this::maybeInvokeCommitCallbacks,
```

Comment:
@philipnee I moved the invocation of the callbacks here, after closing the background thread, to avoid missing a callback that is still completed before the shutdown of the background thread. LMK if you had a strong reason to invoke the callbacks the last time before stopping the network interaction.